xwiki-platform-core/xwiki-platform-query/xwiki-platform-query-xwql/src/main/java/org/xwiki/query/xwql/internal/hql/PropertyPrinter.java

```diff
@@ -40,14 +42,23 @@ void print(PropertyInfo prop, Printer printer) throws Exception
             if (className != null) {
                 prop.alias = printer.getContext().getAliasGenerator().generate(prop.object.alias + "_" + prop.name);
                 printer.from.append(", ").append(className).append(" as ").append(prop.alias);
+                // Join the property value for lists
```

Comment:
General remark: Try putting the "why" in comments rather than the "what" :) It helps more when reading back the code.